src/CommandController.ts

```diff
@@ -165,14 +170,36 @@ export default class CommandController {
             if (holder.prefix !== prefix) return;
 
             // handlers that register the "*" command will get all commands with that prefix (unless they already have gotten it once)
+
             if (holder.command.aliases.some(x => x === command)) {
-                holder.handler.call(null, message, isAdmin, command, parts.slice(1));
+                if (this.checkCooldown(holder, message)) {
+                    holder.handler.call(null, message, isAdmin, command, parts.slice(1));
+                }
             } else if (holder.command.aliases.some(x => x === "*")) {
-                holder.handler.call(null, message, isAdmin, "*", Array<string>().concat(command, parts.slice(1)));
+                if (this.checkCooldown(holder, message)) {
+                    holder.handler.call(null, message, isAdmin, "*", Array<string>().concat(command, parts.slice(1)));
+                }
             }
         });
     }
 
+    private checkCooldown(holder: CommandHolder, message: Discord.Message): boolean {
+        if (holder.cooldown) {
+            const last = holder.lastUsed;
+            const wait = holder.cooldown;
+            const now = Date.now();
+            const remaining = last + wait - now;
+
+            if (last + wait - now > 0) {
+                message.author.send(`This command is currently on cooldown (${remaining / 1000} seconds remaining)`);
```

Comment:
Might want to round this (`(remaining / 1000).toFixed(decimalPlaces)`)? IEEE754 might produce some nice 15 digit decimals if you're unlucky.